### ISSUE TITLE
multi label classification

### ISSUE THREAD

**Issue by mks0601** [rykov8/ssd_keras#6]
hi i have a question about assigning boxes to gt

In matching step, paper firstly match each gt to the default box with maximum iou(1 default box per gt)
Second, they assigned a default box whose iou is larger than 0.5 with any gt. So, did you assign a default box to 'multi class'? or just 'one class' with maximum iou?

for example, a default box B has iou with aeroplane 0.6, person 0.7 in an image. then did you assign that box B to the only person? or both of aeroplane and person?

**Comment by rykov8** [comment]
@mks0601  Hi! Actually, I forgot about matching each gt to the default box with maximum iou(1 default box per gt), because for my problem the set of priors is dense enough to have at least one prior, that has iou greater than 0.5 (actually, quite often I get 15-20 priors matched). But I'd better also fix this thing, thanks. Your comments are very usefull!
As for multi label: I decided to use the first option, namely if a default box B has iou with aeroplane 0.6, person 0.7 in an image, then I assign that box B to the person class only. I didn't check this issue with the original code though. However, I believe, that the authors also use this strategy, al least, because there is softmax activation for every prior, that doesn't imply mutli label classification. Moreover, if this thing happens, I believe, that some neighbouring priors will simply become responsible for one object (e.g. aeroplane) and others for another object (e.g. person).

**Comment by mks0601** [comment]
Thank you for answering.

did you ever tried your implementation on PASCAL VOC? i`m trying, however because of the unbalanced number between classes, my model do not correctly classify the objects.(there are many 'person' class in pascal voc). any comment to help?

**Comment by rykov8** [comment]
No, I didn't try to test my implementation on PASCAL, because I try to solve another problem using real-time detector. However, I have two suggestions, how to avoid imbalance here. The first one is to try to avoid imbalance during mini-batch sampling procedure. The second one, and in my view a better way, is to weight classes in categorical loss. You can refer for example to [sklearn log-loss](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html#sklearn.metrics.log_loss) (I believe, that in sources they implement both binary and mutliclass log-loss) and also `Keras` method `fit` of [`Model` class](https://keras.io/models/model/#methods) has such option. The idea is very simple yet quite useful when you face imbalanced data.
